File: backend/app/models/db_models/task.py
from enum import Enum
from typing import List, Optional
from app.models.db_models.tree import Tree
from app.models.db_models.file import File
from app.models.db_models.base_entity import BaseEntity
from app.models.db_models.tree_decision import TreeDecision


class TaskType(str, Enum):
    PROBLEM = "problem",
    GOAL = "goal",
    HYPOTHESIS = "hypothesis"


class Task(BaseEntity):
    title: str
    user_description: str
    type: TaskType
    company_id: str
    include_company_context: bool = False
    trees: Optional[List[Tree]] = []
    files: Optional[List[File]] = []
    root_cause_analysis: Optional[TreeDecision] = None
    how_trees_decisions: Optional[List[TreeDecision]] = None
    hypotheses_tested: Optional[List[TreeDecision]] = None

File: frontend/src/utils/types/task.ts
import { Tree } from './tree';
import { File } from './file';
import { TreeDecision } from './tree_decision';

export enum TaskType {
  PROBLEM = 'problem',
  GOAL = 'goal',
  HYPOTHESIS = 'hypothesis',
}

export interface Task extends BaseEntity {
  title: string;
  user_description: string;
  type: TaskType;
  company_id: string;
  trees: Tree[];
  files: File[];
  root_cause_analysis: TreeDecision | null;
  hypotheses_tested: TreeDecision[] | null;
  how_trees_decisions: TreeDecision[] | null;
}
